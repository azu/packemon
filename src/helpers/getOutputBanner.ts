import { BundleBuild } from '../types';

export function getOutputBanner({ platform, support, format }: BundleBuild): string {
  return [
    '// Generated with Packemon: https://packemon.dev\n',
    `// Platform: ${platform}, Support: ${support}, Format: ${format}\n\n`,
  ].join('');
}
