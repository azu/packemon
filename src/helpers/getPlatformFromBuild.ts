import { Format, Build, Platform } from '../types';

export default function getPlatformFromBuild(format: Format, build: Build): Platform {
  if (format === 'cjs' || format === 'mjs') {
    return 'node';
  }

  if (format === 'esm' || format === 'umd') {
    return 'browser';
  }

  // "lib" is a shared format across all platforms,
  // and when a package wants to support multiple platforms,
  // we must down-level the "lib" format to the lowest platform.
  if (build.flags.requiresSharedLib) {
    const platforms = new Set(build.platforms);

    if (platforms.has('browser')) {
      return 'browser';
    } else if (platforms.has('node')) {
      return 'node';
    }
  }

  return build.platforms[0];
}
