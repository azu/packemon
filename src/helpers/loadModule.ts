export function loadModule(name: string, message: string): unknown {
  try {
    // eslint-disable-next-line
    return require(name);
  } catch {
    throw new Error(`${message} Please install with \`yarn add --dev ${name}\`.`);
  }
}
