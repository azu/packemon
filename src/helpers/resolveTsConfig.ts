import { Path, parseFile } from '@boost/common';
import { TSConfigStructure } from '../types';

export default function resolveTsConfig(path: Path): TSConfigStructure {
  const contents = parseFile<TSConfigStructure>(path);

  // TODO deep merge
  if (contents.extends) {
    return {
      ...resolveTsConfig(path.parent().append(contents.extends)),
      ...contents,
    };
  }

  return contents;
}
