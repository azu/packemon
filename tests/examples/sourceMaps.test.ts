import { testExampleOutput } from '../helpers';

describe('Source maps', () => {
  testExampleOutput('source-maps.ts');
});
