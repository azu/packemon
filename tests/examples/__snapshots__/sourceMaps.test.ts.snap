// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source maps transforms example test case 1`] = `
Array [
  "/cjs/index-node-current-cjs.cjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: cjs
'use strict';
//# sourceMappingURL=index-node-current-cjs.cjs.map
",
]
`;

exports[`Source maps transforms example test case 2`] = `
Array [
  "/cjs/index-node-experimental-cjs.cjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: cjs
'use strict';
//# sourceMappingURL=index-node-experimental-cjs.cjs.map
",
]
`;

exports[`Source maps transforms example test case 3`] = `
Array [
  "/cjs/index-node-legacy-cjs.cjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: cjs
'use strict';
//# sourceMappingURL=index-node-legacy-cjs.cjs.map
",
]
`;

exports[`Source maps transforms example test case 4`] = `
Array [
  "/cjs/index-node-stable-cjs.cjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: cjs
'use strict';
//# sourceMappingURL=index-node-stable-cjs.cjs.map
",
]
`;

exports[`Source maps transforms example test case 5`] = `
Array [
  "/esm/index-browser-current-esm.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: esm
//# sourceMappingURL=index-browser-current-esm.js.map
",
]
`;

exports[`Source maps transforms example test case 6`] = `
Array [
  "/esm/index-browser-experimental-esm.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: esm
//# sourceMappingURL=index-browser-experimental-esm.js.map
",
]
`;

exports[`Source maps transforms example test case 7`] = `
Array [
  "/esm/index-browser-legacy-esm.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: esm
//# sourceMappingURL=index-browser-legacy-esm.js.map
",
]
`;

exports[`Source maps transforms example test case 8`] = `
Array [
  "/esm/index-browser-stable-esm.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: esm
//# sourceMappingURL=index-browser-stable-esm.js.map
",
]
`;

exports[`Source maps transforms example test case 9`] = `
Array [
  "/lib/index-browser-current-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: lib
'use strict';
//# sourceMappingURL=index-browser-current-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 10`] = `
Array [
  "/lib/index-browser-experimental-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: lib
'use strict';
//# sourceMappingURL=index-browser-experimental-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 11`] = `
Array [
  "/lib/index-browser-legacy-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: lib
'use strict';
//# sourceMappingURL=index-browser-legacy-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 12`] = `
Array [
  "/lib/index-browser-stable-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: lib
'use strict';
//# sourceMappingURL=index-browser-stable-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 13`] = `
Array [
  "/lib/index-native-current-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: native, Support: current, Format: lib
'use strict';
//# sourceMappingURL=index-native-current-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 14`] = `
Array [
  "/lib/index-native-experimental-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: native, Support: experimental, Format: lib
'use strict';
//# sourceMappingURL=index-native-experimental-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 15`] = `
Array [
  "/lib/index-native-legacy-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: native, Support: legacy, Format: lib
'use strict';
//# sourceMappingURL=index-native-legacy-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 16`] = `
Array [
  "/lib/index-native-stable-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: native, Support: stable, Format: lib
'use strict';
//# sourceMappingURL=index-native-stable-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 17`] = `
Array [
  "/lib/index-node-current-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: lib
'use strict';
//# sourceMappingURL=index-node-current-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 18`] = `
Array [
  "/lib/index-node-experimental-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: lib
'use strict';
//# sourceMappingURL=index-node-experimental-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 19`] = `
Array [
  "/lib/index-node-legacy-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: lib
'use strict';
//# sourceMappingURL=index-node-legacy-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 20`] = `
Array [
  "/lib/index-node-stable-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: lib
'use strict';
//# sourceMappingURL=index-node-stable-lib.js.map
",
]
`;

exports[`Source maps transforms example test case 21`] = `
Array [
  "/mjs/index-node-current-mjs.mjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: mjs
//# sourceMappingURL=index-node-current-mjs.mjs.map
",
]
`;

exports[`Source maps transforms example test case 22`] = `
Array [
  "/mjs/index-node-experimental-mjs.mjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: mjs
//# sourceMappingURL=index-node-experimental-mjs.mjs.map
",
]
`;

exports[`Source maps transforms example test case 23`] = `
Array [
  "/mjs/index-node-legacy-mjs.mjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: mjs
//# sourceMappingURL=index-node-legacy-mjs.mjs.map
",
]
`;

exports[`Source maps transforms example test case 24`] = `
Array [
  "/mjs/index-node-stable-mjs.mjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: mjs
//# sourceMappingURL=index-node-stable-mjs.mjs.map
",
]
`;

exports[`Source maps transforms example test case 25`] = `
Array [
  "/package.json",
  Object {
    "files": Array [
      "cjs/",
      "esm/",
      "lib/",
      "mjs/",
      "src/",
      "umd/",
    ],
    "name": "examples",
  },
]
`;

exports[`Source maps transforms example test case 26`] = `
Array [
  "/umd/index-browser-current-umd.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: umd
//# sourceMappingURL=index-browser-current-umd.js.map
",
]
`;

exports[`Source maps transforms example test case 27`] = `
Array [
  "/umd/index-browser-experimental-umd.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: umd
//# sourceMappingURL=index-browser-experimental-umd.js.map
",
]
`;

exports[`Source maps transforms example test case 28`] = `
Array [
  "/umd/index-browser-legacy-umd.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: umd
//# sourceMappingURL=index-browser-legacy-umd.js.map
",
]
`;

exports[`Source maps transforms example test case 29`] = `
Array [
  "/umd/index-browser-stable-umd.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: umd
//# sourceMappingURL=index-browser-stable-umd.js.map
",
]
`;
